test/ex_machina/ecto_test.exs

```diff
@@ -111,5 +171,34 @@ defmodule ExMachina.EctoTest do
     comment = MyApp.EctoFactories.create(:comment, article: my_article)
 
     assert comment.article == my_article
+    assert_received {:created, %{model: %MyApp.Comment{}} = comment_changeset}
+    # belongs_to association should not be handled by changeset
+    assert comment_changeset.changes[:article] == nil
+  end
+
+  test "create/2 raises unless factory returns Ecto.Model" do
+    assert_raise ArgumentError, ~r"not Ecto model", fn ->
+      MyApp.EctoFactories.create(:user_map)
+    end
+  end
+
+  test "create/2 creates model with `has_many` associations" do
+    article = MyApp.EctoFactories.create(:article_with_comments)
+
+    assert %{comments: [%{body: "This is great!"}]} = article
+    assert_received {:created, %{model: %MyApp.Article{}} = article_changeset}
+    assert %{changes: %{comments: [%{action: :insert}]}} = article_changeset
+    refute_received {:created, %{model: %MyApp.Comment{}}}
```

Comment:
Comments are already created with `article_changeset` (this is handled by Ecto). Individual comments should not be passed to `Repo.insert!` again - this would lead to duplicated entries.